client/app/bundles/course/assessment/pages/AssessmentGenerate/utils.ts

```diff
@@ -37,111 +54,90 @@ export function extractQuestionPrototypeData(
         },
       },
     },
   };
 }
 
+type IndexedTestCase = Record<
+  number,
+  { expression: string; expected: string; hint: string }
+>;
+
 export const buildGenerateRequestPayload = (
   codaveriData: CodaveriGenerateFormData,
   questionData: QuestionPrototypeFormData,
 ): FormData => {
   const data = new FormData();
-  // TODO: Currently we are injecting the existing question data into the custom prompt directly.
-  // When Codaveri implements this as a feature, make sure to use the updated request model.
-  const fragments = [codaveriData.customPrompt];
+  const isFirstGeneratedQuestion =
+    JSON.stringify(questionData) === JSON.stringify(defaultQuestionFormData);
+
+  const populateTestCases = (
+    fieldName: string,
+    testCases: MetadataTestCase[],
+    testCaseDict: IndexedTestCase,
+  ): void => {
+    if (testCases && testCases.length > 0) {
+      testCases.forEach((testCase, index) => {
+        testCaseDict[index + 1] = {
+          expression: testCase.expression,
+          expected: testCase.expected,
+          hint: testCase.hint,
+        };
+      });
+
+      data.append(fieldName, JSON.stringify(testCaseDict));
+    }
+  };
+
+  const publicTestCases = {} as IndexedTestCase;
+  const privateTestCases = {} as IndexedTestCase;
+  const evaluationTestCases = {} as IndexedTestCase;
+
+  data.append(
+    'is_first_generated_question',
+    isFirstGeneratedQuestion.toString(),
+  );
+
   if (questionData?.question?.title) {
-    fragments.push(`title is currently "${questionData?.question?.title}"`);
+    data.append('title', questionData.question.title);
   }
 
   if (questionData?.question?.description) {
-    fragments.push(
-      `description is currently"${questionData?.question?.description}"`,
-    );
+    data.append('description', questionData.question.description);
   }
 
   if (questionData?.testUi?.metadata?.solution) {
-    fragments.push(
-      `solution is currently"${questionData?.testUi?.metadata?.solution}"`,
-    );
+    data.append('solution', questionData.testUi.metadata.solution);
   }
 
   if (questionData?.testUi?.metadata?.submission) {
-    fragments.push(
-      `template is currently"${questionData?.testUi?.metadata?.submission}"`,
-    );
+    data.append('template', questionData.testUi.metadata.submission);
   }
 
-  if (
-    questionData?.testUi?.metadata?.testCases?.public &&
-    questionData?.testUi?.metadata?.testCases?.public.length > 0
-  ) {
-    const subfragments = [`The current public test cases are:`];
-    questionData?.testUi?.metadata?.testCases?.public.forEach(
-      (testCase, index) => {
-        fragments.push(
-          `${index + 1}. ${testCase.expression} is ${testCase.expected} (${testCase.hint})`,
-        );
-      },
-    );
-    fragments.push(subfragments.join('\n'));
-  }
+  populateTestCases(
+    'public_test_cases',
+    questionData?.testUi?.metadata?.testCases?.public,
+    publicTestCases,
+  );
```

Comment:
Thanks for updating the duplicated code into the function.

Nit: personally I would write like this:
```ts
if (questionData?.testUi?.metadata?.testCases?.public?.length > 0) {
  populateTestCases(
    'public_test_cases',
    questionData?.testUi?.metadata?.testCases?.public,
    {} as IndexedTestCase,
  );
}
```
in order to follow the style of the earlier lines of code, and also skip definitions of the `const xxxTestCases`

Reply:
done